File: backend/routes/faces.js
const express = require('express');
const router = express.Router();
const Face = require('../models/Face');

// @route   GET /api/family-faces
// @desc    Get all family face metadata
router.get('/', async (req, res) => {
  try {
    const faces = await Face.find().sort({ createdAt: -1 });
    res.json(faces);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// @route   POST /api/family-faces
// @desc    Add a new family face
router.post('/', async (req, res) => {
  try {
    const { name, imageUrl } = req.body;
    if (!name || !imageUrl) {
      return res.status(400).json({ message: 'Name and Image URL are required' });
    }

    const newFace = new Face({ name, imageUrl });
    const savedFace = await newFace.save();
    res.status(201).json(savedFace);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// @route   DELETE /api/family-faces/:id
// @desc    Delete a family face record
router.delete('/:id', async (req, res) => {
  try {
    const face = await Face.findByIdAndDelete(req.params.id);
    if (!face) return res.status(404).json({ message: 'Face record not found' });
    res.json({ message: 'Face record deleted successfully', id: req.params.id });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

module.exports = router;

File: backend/routes/faces.py
from fastapi import APIRouter, HTTPException, Depends, status
from bson import ObjectId
from datetime import datetime
from database import get_db
from models import FaceCreate, serialize_mongo_doc

router = APIRouter(prefix="/api/family-faces", tags=["faces"])

@router.get("")
async def get_faces(db=Depends(get_db)):
    try:
        cursor = db.faces.find().sort("createdAt", -1)
        faces = await cursor.to_list(length=1000)
        return [serialize_mongo_doc(face) for face in faces]
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.post("", status_code=201)
async def create_face(payload: FaceCreate, db=Depends(get_db)):
    try:
        if not payload.name or not payload.imageUrl:
            raise HTTPException(status_code=400, detail="Name and Image URL are required")
            
        face_doc = {
            "name": payload.name,
            "imageUrl": payload.imageUrl,
            "createdAt": datetime.utcnow()
        }
        
        result = await db.faces.insert_one(face_doc)
        face_doc["_id"] = result.inserted_id
        return serialize_mongo_doc(face_doc)
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.delete("/{id}")
async def delete_face(id: str, db=Depends(get_db)):
    try:
        if not ObjectId.is_valid(id):
            raise HTTPException(status_code=400, detail="Invalid face record ID format")
            
        result = await db.faces.find_one_and_delete({"_id": ObjectId(id)})
        if not result:
            raise HTTPException(status_code=404, detail="Face record not found")
            
        return {"message": "Face record deleted successfully", "id": id}
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
